Bintray cleanup

abesto opened the pull request:
As discussed in #2, this script finds old versions of Bintray packages and deletes them. Looks a bit like this:

![snip 2019-07-10 190653](https://user-images.githubusercontent.com/59982/60993219-e6194f00-a345-11e9-8384-ce4eb2f100ee.png)

If you want to try it out, make sure you have Python 3.7 and [Poetry](https://github.com/sdispater/poetry) (`curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python`), then:

```sh
cd bintray-cleanup
export BINTRAY_API_KEY=...
export BINTRAY_USERNAME=...
poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365
```

This lists all versions of `zipkin-dependencies` older than 365 days. Replace `list-old-versions` with `delete-old-versions` to delete them. The `delete-old-versions` command defaults to dry-run mode; this can be disabled with `--no-dryrun`. I didn't actually pull the trigger on any actual version deletions pending review of this code and @adriancole explicitly calling down hellfire on a test version of a low-impact package.

My suggested approach is:

```sh
# Verify things look mostly sane; probably want to Ctrl-C after a few versions
poetry run bintray_cleanup/main.py delete-old-versions openzipkin maven zipkin-dependencies 365

# Being very very careful, delete the oldest version
poetry run bintray_cleanup/main.py delete-old-versions openzipkin maven zipkin-dependencies 365 \
    --no-dryrun --limit 1

# Verify the version is gone (non-dry-run delete-old-versions flushes the response cache)
poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365

# Still being careful, start deleting many versions, verifying each manually.
# Probably want to Ctrl-C after a couple of versions.
poetry run bintray_cleanup/main.py delete-old-versions openzipkin maven zipkin-dependencies 365 \
    --no-dryrun

# Aborted runs don't clear the cache, so let's do that manually
poetry run bintray_cleanup/main.py clear-cache

# Verify all is well
poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365

# Still being careful, maybe at this point we're confident enough to
# drop per-version manual verification, do a few versions first
poetry run bintray_cleanup/main.py delete-old-versions openzipkin maven zipkin-dependencies 365 \
    --no-dryrun --limit 3 --yes

# Again, verify all is well
poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365

# I'M FEELING LUCKY
poetry run bintray_cleanup/main.py delete-old-versions openzipkin maven zipkin-dependencies 365 \
    --no-dryrun --yes
```

abesto committed:
Step one: list versions, find old ones

abesto committed:
bintray_cleanup is now feature complete I think

abesto committed:
Blacken

codefromthecrypt commented:
cool! I will look at this by saturday if not before. feel free to proceed regardless

codefromthecrypt commented:
something silly I suspect..

```bash
$ poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365
Traceback (most recent call last):
  File "bintray_cleanup/main.py", line 6, in <module>
    import click
ModuleNotFoundError: No module named 'click'

$ pip install click
Requirement already satisfied: click in /usr/local/lib/python3.7/site-packages (7.0)
```

abesto commented:
Weird. Try running poetry install first. I expected run to take care of it

On Sat, Jul 13, 2019, 11:39 Adrian Cole <notifications@github.com> wrote:

> something silly I suspect..
>
> $ poetry run bintray_cleanup/main.py list-old-versions openzipkin maven zipkin-dependencies 365
> Traceback (most recent call last):
>   File "bintray_cleanup/main.py", line 6, in <module>
>     import click
> ModuleNotFoundError: No module named 'click'
>
> $ pip install click
> Requirement already satisfied: click in /usr/local/lib/python3.7/site-packages (7.0)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAOUTW34DFFKNMNASVAW33P7GWG3A5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZ3PEMY#issuecomment-511111731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAOUTSDYYN6K4XGFSUL24LP7GWG3ANCNFSM4H7R6U3Q>
> .
>

codefromthecrypt commented:
eheh I didn't ever run poetry install

On Sat, Jul 13, 2019 at 8:21 PM Zoltán Nagy <notifications@github.com>
wrote:

> Weird. Try running poetry install first. I expected run to take care of it
>
> On Sat, Jul 13, 2019, 11:39 Adrian Cole <notifications@github.com> wrote:
>
> > something silly I suspect..
> >
> > $ poetry run bintray_cleanup/main.py list-old-versions openzipkin maven
> zipkin-dependencies 365
> > Traceback (most recent call last):
> > File "bintray_cleanup/main.py", line 6, in <module>
> > import click
> > ModuleNotFoundError: No module named 'click'
> >
> > $ pip install click
> > Requirement already satisfied: click in
> /usr/local/lib/python3.7/site-packages (7.0)
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAOUTW34DFFKNMNASVAW33P7GWG3A5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZ3PEMY#issuecomment-511111731
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAAOUTSDYYN6K4XGFSUL24LP7GWG3ANCNFSM4H7R6U3Q
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAPVV6TMS6PRZQX4N6EDQ3P7G3FJA5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZ3PXSI#issuecomment-511114185>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAPVVYBVNYU2D3SXGJ7IZ3P7G3FJANCNFSM4H7R6U3Q>
> .
>

abesto commented:
@adriancole lacking time to run this, if you're feeling trusting, point me at a repo and package age cutoff (maybe even a "this is a version to keep for sure"), and I'll do the thing.

codefromthecrypt commented:
I trust you and you can point to any repo. If feeling cautious you can do the cloud ones first (zipkin-aws and zipkin-gcp) thanks!

abesto committed:
Print timestamp before executing deletion

abesto committed:
Delete just the version we want, not the full package ;)

abesto committed:
Fix off-by-one error in --limit default value

abesto commented:
Happy to report `zipkin-aws` is down to 14 versions (killed forty-something old versions). Full recording of interactive terminal session via `script` is at https://gist.github.com/abesto/aad821ad71f5628915cb3ecbacbe81ef (`cat` it in a terminal to have it pretty-printed, contains ANSI escapes, sorry).

Methinks it's prudent to wait a few days, maybe a week, to make sure any unexpected side-effects are recognized. Then move on to the next, I guess? :)

\o/

codefromthecrypt commented:
That's super cool. As our authoritative are maven central, I think we haven't risk moving forward. Though I am not sure who would be using jcenter instead of maven central. In the worst case, we can always re-publish off the tag. I'd move forward, but understand if you prefer to wait a bit.

abesto commented:
Alright, you've convinced me. So we want cleanup of package versions older
than 365 days across all Maven repos under the openzipkin org on Bintray.
Confirm?

On Tue, Jul 23, 2019, 00:53 Adrian Cole <notifications@github.com> wrote:

> That's super cool. As our authoritative are maven central, I think we
> haven't risk moving forward. Though I am not sure who would be using
> jcenter instead of maven central. In the worst case, we can always
> re-publish off the tag. I'd move forward, but understand if you prefer to
> wait a bit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAOUTWAMLWLHTBBH6D63O3QAZB7LA5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2RPW7Y#issuecomment-513997695>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAOUTVG4CWZRHD6BSKE2QTQAZB7LANCNFSM4H7R6U3Q>
> .
>

codefromthecrypt commented:
roger Wilco!

On Tue, Jul 23, 2019, 4:32 PM Zoltán Nagy <notifications@github.com> wrote:

> Alright, you've convinced me. So we want cleanup of package versions older
> than 365 days across all Maven repos under the openzipkin org on Bintray.
> Confirm?
>
> On Tue, Jul 23, 2019, 00:53 Adrian Cole <notifications@github.com> wrote:
>
> > That's super cool. As our authoritative are maven central, I think we
> > haven't risk moving forward. Though I am not sure who would be using
> > jcenter instead of maven central. In the worst case, we can always
> > re-publish off the tag. I'd move forward, but understand if you prefer to
> > wait a bit.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAOUTWAMLWLHTBBH6D63O3QAZB7LA5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2RPW7Y#issuecomment-513997695
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAAOUTVG4CWZRHD6BSKE2QTQAZB7LANCNFSM4H7R6U3Q
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAPVV7JHP6LUOMS6V7AWELQA26YXA5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2SLFAQ#issuecomment-514110082>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAPVV3QK5ZRJLMFUWVIFYLQA26YXANCNFSM4H7R6U3Q>
> .
>

abesto committed:
Add commands to list, delete across all packages in a repo

abesto commented:
Deleted 331 versions so far (each deletion takes several seconds, but don't want to parallelize, this is fine for now). Will continue tomorrow. According to https://api.bintray.com/orgs/openzipkin which shows quota information to org admins, we are now at ~6 GB used, so we should fit nicely into even the default storage quota. Deleted versions so far: 

<details><pre>
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.4.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.4.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.4.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.4.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.5.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.6.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.9.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.9.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.9.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.10.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.10.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.10.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.10.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.11.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.12.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.12.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.12.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.13.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.14.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.15.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.15.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.16.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.16.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.17.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.17.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.18.6
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.19.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.20.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.20.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.21.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.21.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.21.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.21.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.21.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.1.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.2.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.4.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.4.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.4.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.4.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.5.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/0.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/centralsync-maven-plugin/versions/0.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.9.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.11.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.11.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.13.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.14.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.14.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.15.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.15.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.15.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.15.4
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.16.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.17.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.17.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/3.18.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.4
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.5
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.0.6
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.3.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.3.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.3.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.3.4
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.5.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.5.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.7.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.7.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.8.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.9.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.9.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.10.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.11.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.12.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.4
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.14.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.14.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.14.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.14.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.15.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.15.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.16.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.16.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.16.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.17.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.17.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.17.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.18.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.18.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.18.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.19.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.19.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.19.2
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.19.3
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/5.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/5.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/4.13.5
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/5.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/brave/versions/5.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.0.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.0.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.0.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.0.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.3.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.3.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.3.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.5.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.5.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.5.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.5.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.6.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.7.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.7.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.8.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.9.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.9.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.9.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.10.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.10.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.10.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.10.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.11.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.11.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.11.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.11.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.9.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.11.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.12.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/1.12.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-dependencies/versions/2.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.2.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.3.6
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/0.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.0.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.6
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/1.2.7
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-finagle/versions/2.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.4.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.4.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.4.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.4.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.5.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.5.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.6
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.7
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.8
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.9
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.10
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.11
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.12
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.13
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.6.14
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.7.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/0.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.0.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.0.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.0.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.0.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.0.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.0.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.1.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.1.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/1.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.1.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.1.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.1.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.2.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.2.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.2.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.2.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.3.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.3.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.3.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.3.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.4.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.4.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.5.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.6.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.6.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.5
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin-reporter-java/versions/2.7.6
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.8.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.9.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.10.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.11.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.11.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.12.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.12.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.13.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.13.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.14.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.14.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.14.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.14.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.14.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.15.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.16.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.16.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.16.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.17.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.17.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.18.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.19.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.19.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.19.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.19.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.19.4
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.20.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.20.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.21.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.22.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.22.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.23.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.23.1
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.23.2
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.23.3
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.24.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.25.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.26.0
DELETE https://api.bintray.com/packages/openzipkin/maven/zipkin/versions/1.26.1
</pre></details>

Full log: https://gist.github.com/abesto/daa2937fef5a27b453e2ecfe28a44a88

codefromthecrypt commented:
420 time!

On Wed, Jul 24, 2019 at 3:14 AM Zoltán Nagy <notifications@github.com>
wrote:

> FTR,: deletion of ~420 versions is now in progress. Will update this
> comment once done.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAPVV4BTOEQIUQDWBL3I4DQA5C67A5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2T7IXA#issuecomment-514323548>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAPVV6DJGQZ3OJOYHOUHN3QA5C67ANCNFSM4H7R6U3Q>
> .
>

abesto commented:
Done and done!

```
Cutoff date 365 days ago: 2018-07-24 20:30:56.467496+00:00
Found 0 versions created BEFORE 2018-07-24 20:30:56.467496+00:00:

Found 108 versions created AFTER 2018-07-24 20:30:56.467496+00:00:
  zipkin: 2.10.4 2.11.1 2.11.2 2.11.3 2.11.4 2.11.5 2.11.6 2.11.7 2.11.8 2.11.12 2.12.1 2.12.2 2.11.13 2.12.6 2.12.7 2.12.8 2.12.9 2.12.10 2.14.1 2.14.2 2.15.0
  zipkin-reporter-java: 2.7.7 2.7.8 2.7.9 2.7.10 2.7.11 2.7.12 2.7.13 2.7.14 2.7.15 2.8.0 2.8.1 2.8.2 2.8.3 2.8.4 2.9.0 2.10.0
  brave: 5.1.4 5.1.5 5.2.0 5.3.0 5.3.1 5.3.2 5.3.3 4.13.6 5.4.0 5.4.1 5.4.2 5.4.3 5.4.4 5.5.0 5.5.1 5.5.2 5.6.0 5.6.1 5.6.2 5.6.3 5.6.8 5.6.9
  zipkin-gcp: 0.6.4 0.7.0 0.7.1 0.8.0 0.8.1 0.8.2 0.8.3 0.9.0 0.10.0 0.10.1 0.10.2 0.10.3 0.11.0 0.11.1 0.12.0 0.13.0 0.13.1
  brave-cassandra: 0.8.1 0.9.0 0.10.0 0.10.1 0.10.2 0.10.3
  zipkin-dependencies: 2.0.1 2.0.2 2.0.3 2.0.4 2.0.5 2.0.6 2.0.7 2.0.8 2.1.0 2.3.0 2.3.1
  zipkin-layout-factory: 0.0.3 0.0.4 0.0.5
  zipkin-finagle: 2.0.8 2.0.9 2.0.11 2.0.13 2.0.14 2.0.15 2.0.16 2.0.17
  zipkin-proto3: 0.2.0 0.2.1
  centralsync-maven-plugin: 0.1.1
  brave-karaf: 0.1.2
```

Deletion log of this second batch (89 versions): https://gist.github.com/abesto/9bb7c3c0b7220a9f224320fbf0b6c44e

Currently we use about 2.7 GB storage on Bintray \o/

I'm gonna merge this, it doesn't get any more production than this.

codefromthecrypt commented:
this is fantastic. next time I see jfrog folks I dont have to pull hat down
in embarrassment:)

great job!

On Thu, Jul 25, 2019, 4:33 AM Zoltán Nagy <notifications@github.com> wrote:

> Merged #3 <https://github.com/openzipkin/zipkin-release/pull/3> into
> master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin-release/pull/3?email_source=notifications&email_token=AAAPVV6WFRHYRFBXAKGOBV3QBC4ATA5CNFSM4H7R6U32YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOSV3GWQI#event-2507565889>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAPVV2JVMH4GTFYB225AS3QBC4ATANCNFSM4H7R6U3Q>
> .
>
